be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -89,11 +91,11 @@ bool ScanOperator::has_output() const {
     if (!_get_scan_status().ok()) {
         return true;
     }
-
-    for (const auto& chunk_source : _chunk_sources) {
-        if (chunk_source != nullptr && chunk_source->has_output()) {
-            return true;
-        }
+    if (has_shared_chunk_source()) {
+        return true;
+    }
```

Comment:
Maybe we needn't return true when `has_shared_chunk_source` is true?
`ScanOperator` can be executed, only when it has ready chunk or it can commit next i/o task.